pyopensprinkler/__init__.py

```diff
@@ -116,13 +116,18 @@ def _refresh_state(self):
         (_, content) = self.request("/ja")
         self._state = content
 
+    def _retrieve_state(self):
+        if self._state == None:
+            raise OpenSprinklerNoStateError("No state. Please refresh")
```

Comment:
I considered doing a refresh here but decided against it as it will do an unexpected I/O